### PR TITLE
Secondary Navbar

### LINKED ISSUE #6: Secondary Sticky Nav Bar with Anchor and Scroll
## Description

A Navigation Bar that sticks on the top and navigates users to different sections on the page.

image

## Learning Objectives

- React-Scroll
- Tailwind CSS (lots of CSS!) 

## Todo

These points are a rough guideline. Please feel free to discuss with others on the team about the best way to design this component!

- [ ] Create a Nav Bar component with for the home page. Note that: 
  - Nav items should be "active" when a section is in viewport 
  - Clicking on the item will bring users to a section on page, there should be scroll animations
  - [React Scroll](https://github.com/fisshy/react-scroll) is recommended, but feel free to explore other methods!
- [ ] Style the Nav Bar component to match [Figma Design](https://www.figma.com/file/PFCGAeZzwLEyzNaovh6Zfq/Toriis?node-id=3%3A1243&t=94OBWIFhKXQRQRI3-0)
- [ ] Extract the Nav Bar component and make it adaptable to multiple pages 

Additional notes: this issue ticket will go concurrently with other tickets on building out the sections, so the sections may not be ready when you start this ticket, you should create temporary sections to test the functionality.

## Useful Links

- [Figma Design](https://www.figma.com/file/PFCGAeZzwLEyzNaovh6Zfq/Toriis?node-id=3%3A1243&t=94OBWIFhKXQRQRI3-0)
- [React Scroll](https://github.com/fisshy/react-scroll)
- [React Scroll Example Code Sandbox](https://codesandbox.io/s/basic-6t84k)
- [CSS Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)

## Timeline 

By beginning of Monday dev night

### PR DESCRIPTION
**Pull request by eeshabarua**
## Summary

Creating a cool sticky nav bar to scroll to different parts of the page!

Closes #6 

## Changes

- Implement React Scroll
- Fix Tiny CSS Things

## Screenshots

https://user-images.githubusercontent.com/61480751/221056150-a2d68e2d-46a2-4c82-8f49-b0fab20abbc9.mov

### CODE REVIEW

**Comment by eeshabarua** [comment]
I'm having the same color issue as Xiaofan, I used the template orange color in the meantime.

**Comment by eeshabarua** [comment]
image
Also, not sure if we want this hover functionality without the anchor. I can delete if not!

File: src/components/SecondaryNavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import type { FC } from 'react'
+import { Link, animateScroll as scroll } from 'react-scroll'
+
+import SampleSection from '../sections/SampleSection'
+
+const NavBar: FC = () => {
```
Comment (zorazrr) [inline src/components/SecondaryNavBar.tsx:6]: Nit. Rename to `SecondaryNavBar`. 

Additionally, because we are reusing this nav bar in other pages as well, it would be great if we can extract things a bit here, so that instead of hard-coding the nav item strings, we could pass them in (maybe as an array of strings) as props!

File: src/info/home.json [hunk 2]
```diff
@@ -1,3 +1,11 @@
 {
-  "example": "an awesome example"
+  "example": "an awesome example",
+  "secondaryNavText": [
```
Comment (zorazrr) [inline src/info/home.json:3]: As mentioned above, we should pass this in as a prop to the secondary nav bar component!

File: src/styles/globals.css [hunk 3]
```diff
@@ -1,3 +1,10 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
+.active {
+  text-decoration: underline;
+  text-decoration-color: orange;
```
Comment (zorazrr) [inline src/styles/globals.css:7]: You mentioned the color issue in comments: color definitions in Tailwind config doesn't transfer into this traditional CSS definition. So what you could do is either hardcoding the hexcode, or define CSS variables as in this [StackOverflow Post](https://stackoverflow.com/questions/64872861/how-to-use-css-variables-with-tailwind-css#:~:text=Armando%27s%20answer%20didn%27t%20work%20for%20me%20but%20with%20this%20change%20it%20did%20work.).

File: src/components/SecondaryNavBar.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          <div className="md:flex">
+            <ul className="flex space-x-8 p-4 text-sm text-white ">
+              <li>
+                <Link
+                  activeClass="active"
+                  to="container1"
+                  spy={true}
+                  smooth={true}
+                  duration={500}
+                  className="no-underline hover:underline hover:underline-offset-8"
```
Comment (zorazrr) [inline src/components/SecondaryNavBar.tsx:20]: Let's give the hover an orange color as well (same as active class styles)! Additionally, we might want to add a [`cursor-pointer`](https://tailwindcss.com/docs/cursor) to indicate that this item is clickable.

**Review by zorazrr — CHANGES_REQUESTED** [review]
The scrolling animation is looking fire!!!! 
One major thing is to make sure you use `yarn` instead of `npm`, so the `package-lock.json` should not be here. Additionally, we want to pass in nav items as props. Other than that, it's looking good, just some small styling changes.

File: src/components/SecondaryNavBar.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+import SampleSection from '../sections/SampleSection'
+
+const NavBar: FC = () => {
+  return (
+    <>
+      <nav className="sticky top-0 justify-evenly bg-darkTeal py-2.5">
+        <div className="container mx-auto">
+          <div className="md:flex">
+            <ul className="flex space-x-8 p-4 text-sm text-white ">
```
Comment (lizageorge) [inline src/components/SecondaryNavBar.tsx:12]: The Figma has these links styled as 18px, but Tailwind's `text-sm` is 14px - could you remove this?
On Zora's suggestion, I had wrapped the whole body in a custom font-size class to be 18px in my branch, so once that's merged in all body text will automatically be 18px.
tldr; you don't need to specify font size here

File: src/components/SecondaryNavBar.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                >
+                  Divestment History
+                </Link>
+              </li>
+            </ul>
+          </div>
+        </div>
+      </nav>
+      <div id="container1" className="pt-16">
+        <SampleSection text="Our Requests" />
```
Comment (lizageorge) [inline src/components/SecondaryNavBar.tsx:75]: adding onto Zora's idea about the props - you could make these divs using that same array of string titles too. that way we can limit hardcoded text as much as possible too.

File: src/components/SecondaryNavBar.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
+import type { FC } from 'react'
```
Comment (lizageorge) [inline src/components/SecondaryNavBar.tsx:1]: I don't think this is needed (unless you decide to pass in props to this component and make an interface for them)

**Review by lizageorge — CHANGES_REQUESTED** [review]

**Comment by lizageorge** [comment]
This is such a small thing, but I noticed this weird bug where the "UIS System" section isn't always correctly selected when clicked on. Instead the section right before it is highlighted, like the scrolling feature is off by a tiny amount. I can't figure out why this is happening to just this section, it very well might just be a quirk of react-scroll

https://user-images.githubusercontent.com/20449375/221498968-a778bb37-87f7-44f9-b876-ad8d3be2a02a.mov

But i really had to fight the thing to find this nit - otherwise incredibly smooth navbar 😎

**Review by zorazrr — APPROVED** [review]
Looks good, love the scroll animation!

File: src/pages/home/index.tsx [hunk 8]
```diff
@@ -1,12 +1,42 @@
 import type { FC } from 'react'
 
 import InstitutionalDivestments from '../../sections/InstitutionalDivestments'
+import SecondaryNavBar from '../../components/Nav/SecondaryNavBar'
+import SampleSection from '../../sections/SampleSection'
 
 const Home: FC = () => {
+  const data = {
+    text: [
+      { path: 'ourRequests', text: 'Our Requests' },
```
Comment (lizageorge) [inline src/pages/home/index.tsx:10]: Just a suggestion for the future - I think you could have kept this array in the info json file, and then refer to it in this component too (just mapping over it to actually init the divs and SampleSections below).

File: src/pages/home/index.tsx [hunk 9]
```diff
@@ -1,12 +1,42 @@
...
+        <div id="refuteUISReponse" className="pt-20">
+          <SampleSection text="Refute UIS Response" />
+        </div>
+        <div id="UIUCSystem" className="pt-20">
+          <SampleSection text="UIUC System" />
+        </div>
+        <div id="divestmentHistory" className="pt-20">
+          <SampleSection text="Divestment History" />
+        </div>
+        <SampleSection text="" />
```
Comment (lizageorge) [inline src/pages/home/index.tsx:36]: If these extra SampleSections were just for spacing while dev, it might be best to remove them before merging.

**Review by lizageorge — APPROVED** [review]
Looks good to me! Left a comment about the sample sections, but it's not worth a re-review, just a small thing :)

File: src/pages/home/index.tsx (hunk 8 above)
Comment (zorazrr) [reply src/pages/home/index.tsx:10]: For this one, we could probably keep it in `home/index` because it is specific to that page. `home.json` is more so for text information.

File: src/pages/home/index.tsx (hunk 9 above)
Comment (zorazrr) [reply src/pages/home/index.tsx:36]: These are kinda necessary for the last section to function, but potentially @eeshabarua we could make use of the `offset` property that comes with react-scroll to do some adjustments here. Check out this conversation here: https://github.com/fisshy/react-scroll/issues/335.
